src/modules/powerrename/lib/PowerRenameManager.cpp

```diff
@@ -903,22 +903,34 @@ DWORD WINAPI CPowerRenameManager::s_regexWorkerThread(_In_ void* pv)
                                     StringCchCopy(sourceName, ARRAYSIZE(sourceName), originalName);
                                 }
 
-                                PWSTR replaceTerm = nullptr;
                                 SYSTEMTIME fileTime = {0};
 
-                                if (!useFileTime || ( SUCCEEDED(spItem->GetTime(&fileTime)) && SUCCEEDED(spRenameRegEx->PutFileTime(fileTime))))
+                                HRESULT hr = S_OK;
+                                if (useFileTime)
+                                {
+                                    hr = spItem->GetTime(&fileTime);
+                                    if (SUCCEEDED(hr))
+                                    {
+                                        hr = spRenameRegEx->PutFileTime(fileTime);
+                                    }
+                                }
+
+                                PWSTR newName = nullptr;
+
+                                if (SUCCEEDED(hr))
                                 {
-                                    PWSTR newName = nullptr;
                                     // Failure here means we didn't match anything or had nothing to match
                                     // Call put_newName with null in that case to reset it
-                                    spRenameRegEx->Replace(sourceName, &newName);
+                                    hr = spRenameRegEx->Replace(sourceName, &newName);    
+                                }
 
-                                    // fileTime is a file specific value, to be removed after Replace operation
-                                    if (useFileTime)
-                                    {
-                                        spRenameRegEx->ResetFileTime();
-                                    }
-                                    
+                                if (SUCCEEDED(hr) && useFileTime)
```

Comment:
This is also not a good approach, because what HRESULT are you checking here? If this is supposed to check 
`hr = spRenameRegEx->Replace(sourceName, &newName);` then it should be done inside the `if` statement where the function is called.